ts/features/LandingScreenMultiBanner/utils/__tests__/LandingBannerPicker.test.tsx

```diff
@@ -0,0 +1,98 @@
+import * as useIO from "../../../../store/hooks";
+import { appReducer } from "../../../../store/reducers";
+import { GlobalState } from "../../../../store/reducers/types";
+import { renderScreenWithNavigationStoreContext } from "../../../../utils/testWrapper";
+import { MESSAGES_ROUTES } from "../../../messages/navigation/routes";
+import { LandingScreenBannerPicker } from "../../LandingScreenBannerPicker";
+import { updateLandingScreenBannerVisibility } from "../../store/actions";
+import { LandingScreenBannerId } from "../../store/reducer";
+import * as SELECTORS from "../../store/selectors";
+
+const MyComponent = ({ callback }: { callback: () => void }) => (
+  <Button title="TEST" testID="button-test" onPress={callback} />
+);
+
+jest.mock("../landingScreenBannerMap", () => ({
+  get landingScreenBannerMap() {
+    return {
+      TESTING: {
+        component: (something: () => void) => (
+          <MyComponent callback={something} />
+        ),
+        isRenderableSelector: jest.fn()
+      },
+      FILLER: {
+        component: (something: () => void) => (
+          <MyComponent callback={something} />
+        ),
+        isRenderableSelector: jest.fn()
+      },
+      DO_NOT_RENDER: {
+        component: (something: () => void) => (
+          <MyComponent callback={something} />
+        ),
+        isRenderableSelector: jest.fn()
+      }
+    };
+  }
+}));
+const bannerId = "TESTING" as unknown as LandingScreenBannerId;
+
+describe("LandingBannerPicker", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
+  });
+  it("should correctly render a banner if passed a valid id", () => {
+    jest
+      .spyOn(SELECTORS, "landingScreenBannerToRenderSelector")
+      .mockImplementation(_ => bannerId);
+    jest.spyOn(useIO, "useIODispatch").mockImplementation(() => jest.fn());
+
+    const component = renderComponent();
+
+    expect(component.toJSON()).toMatchSnapshot("withBanner");
+  });
+
+  it("should not render a banner if no id is passed", () => {
+    jest
+      .spyOn(SELECTORS, "landingScreenBannerToRenderSelector")
+      .mockImplementation(_ => undefined);
+
+    const component = renderComponent();
+
+    expect(component.toJSON()).toMatchSnapshot("noBanner");
+  });
+  it("should correctly dispatch updateLandingScreenBannerVisibility on press", async () => {
+    jest
+      .spyOn(SELECTORS, "landingScreenBannerToRenderSelector")
+      .mockImplementation(_ => bannerId);
+    const testDispatch = jest.fn();
+    jest.spyOn(useIO, "useIODispatch").mockImplementation(() => testDispatch);
+
+    const component = renderComponent();
+    const button = await component.findByTestId("button-test");
+    fireEvent.press(button);
+
+    expect(testDispatch).toHaveBeenCalledWith(
+      updateLandingScreenBannerVisibility({ id: bannerId, enabled: false })
+    );
+  });
+});
+
+function renderComponent(
+  component: React.ReactElement | null = <LandingScreenBannerPicker />
```

Comment:
If we are not using the parameter, just remove it and set it directly inside the functions'body